src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -1510,51 +1512,83 @@ public ResultSet getTables(
         Arrays.toString(types));
 
     return new SnowflakeDatabaseMetaDataQueryResultSet(GET_TABLES, resultSet, statement) {
+
+      // maintain last table which was returned to use this information for next pagination call
+      String lastTable = null;
+
       @Override
       public boolean next() throws SQLException {
         logger.debug("public boolean next()", false);
         incrementRow();
 
-        // iterate throw the show table result until we find an entry
-        // that matches the table name
+        // get the next table
         while (showObjectResultSet.next()) {
-          String tableName = showObjectResultSet.getString(2);
-
-          String dbName;
-          String schemaName;
-          String kind;
-          String comment;
-
-          if (viewOnly) {
-            dbName = showObjectResultSet.getString(4);
-            schemaName = showObjectResultSet.getString(5);
-            kind = "VIEW";
-            comment = showObjectResultSet.getString(7);
-          } else {
-            dbName = showObjectResultSet.getString(3);
-            schemaName = showObjectResultSet.getString(4);
-            kind = showObjectResultSet.getString(5);
-            comment = showObjectResultSet.getString(6);
-          }
+          if (createRow()) return true;
+        }
 
-          if ((compiledTablePattern == null || compiledTablePattern.matcher(tableName).matches())
-              && (compiledSchemaPattern == null
-                  || compiledSchemaPattern.matcher(schemaName).matches())) {
-            nextRow[0] = dbName;
-            nextRow[1] = schemaName;
-            nextRow[2] = tableName;
-            nextRow[3] = kind;
-            nextRow[4] = comment;
-            nextRow[5] = null;
-            nextRow[6] = null;
-            nextRow[7] = null;
-            nextRow[8] = null;
-            nextRow[9] = null;
-            return true;
-          }
+        // if no more results are found, now trigger pagination call
+        // Note: In actuality, instead of show tables command, a pagination call with a token will be invoked here.
+        // For now, a show command is invoked.
+        ResultSet resultSet1 = executeAndReturnEmptyResultIfNotFound(statement, "show /* JDBC:DatabaseMetaData.getTables() */ tables in schema \"JDBC_DB1\".\"JDBC_SCHEMA11\" limit 1 from '" + lastTable + "'", GET_TABLES);
```

Comment:
This won't work everytime, if table list change after getting first set of values.

Reply:
+1 to what Ilesh said. It would be great to see a one pager that first outlines what is it that we're trying to achieve and how we're thinking of doing it. Looking at code is great but it does not give the big picture.